Result/ResultType.swift

```diff
@@ -1,9 +1,20 @@
 //  Copyright (c) 2015 Rob Rix. All rights reserved.
 
+#if swift(>=3.0)
+	public typealias ResultErrorType = ErrorProtocol
+#else
+	public typealias ResultErrorType = ErrorType
+#endif
+
 /// A type that can represent either failure with an error or success with a result value.
 public protocol ResultType {
+	#if swift(>=2.2)
```

Comment:
This can't be parsed prior to Swift 2.2, so it's useless to try to maintain compatibility with previous versions. Like I mentioned in https://github.com/ReactiveCocoa/ReactiveCocoa/pull/2760/files/6f63eb6569772a90f0b4358acb33b93a4f8105b2#discussion_r56916728, it might not be worth the headache of trying to maintain backwards compatibility. There's already a stable version released for Swift 2.1.x, I see no reason to make the codebase support old versions forever.